src/Core/Utils/BijectiveAssociation.hpp

```diff
@@ -16,69 +18,118 @@ template <typename T1, typename T2>
 class BijectiveAssociation
```

Comment:
It is not clear from the name that we have a set. Could we rename it, e.g. :
```suggestion
class BijectiveAssociationSet
```

Reply:
We do not have a set, we have a discrete representation of a bijection from T1 to T2. This bijection is built from a set of key-value pair but is not a set (it does`nt inherit from any implementation of a set. It just uses a set).

Reply:
To remove this confusion, the class documentation could be reworded as 
``` 
/**
 * \brief Bijective association between two sets {keys} and {values} having the same cardinality.
 * The bijection is constructed by giving all the pairs <key, value> such that f(key) = value.
 * \tparam T1 Type of the keys
 * \tparam T2 Type of the values
```

Reply:
Fine for me too